orion/explorer.py

```diff
@@ -114,30 +131,79 @@ def get_pipelines(self, name=None):
     def get_pipeline(self, pipeline):
         try:
             _id = ObjectId(pipeline)
-            return model.Pipeline.last(_id=_id)
+            return model.Pipeline.last(id=_id)
         except InvalidId:
-            return model.Pipeline.last(name=pipeline)
+            found_pipeline = model.Pipeline.last(name=pipeline)
+            if found_pipeline is None:
+                raise ValueError('Pipeline not found: {}'.format(pipeline))
+            else:
+                return found_pipeline
 
     def load_pipeline(self, pipeline):
         LOGGER.info("Loading pipeline %s", pipeline.name)
         return MLPipeline.from_dict(pipeline.mlpipeline)
 
-    def get_dataruns(self, dataset=None, pipeline=None):
+    def run_experiment(self, name, project, pipeline, dataset, user_id=None):
```

Comment:
Can you make this the last method on the class (just move it down)?
The rationale behind it is that it uses other methods from the class which are defined further down, and all the code is sorted following the "always try to define methods before the other methods that use them" principle.